src/workerd/api/global-scope.c++

```diff
@@ -248,6 +248,19 @@ kj::Promise<DeferredProxy<void>> ServiceWorkerGlobalScope::request(
     body = Body::ExtractedBody(jsStream.addRef());
   }
 
+  // If the request doesn't specify "Content-Length" or "Transfer-Encoding", set "Content-Length"
+  // to the body length if it's known. This ensures handlers for worker-to-worker requests can
+  // access known body lengths if they're set, without buffering bodies.
+  if (method != kj::HttpMethod::HEAD && method != kj::HttpMethod::GET &&
```

Comment:
GET can have a body. Instead of checking the method, maybe what you want is `body != nullptr`?